Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -284,6 +284,39 @@ def check_service_command(formula)
     "Service command does not exist" unless File.exist?(formula.service.command.first)
   end
 
+  def check_cpuid_instruction(formula)
+    return unless formula.prefix.directory?
+    # TODO: add methods to `utils/ast` to allow checking for method use
+    return unless formula.path.read.include? "ENV.runtime_cpu_detection"
+    # Checking for `cpuid` only makes sense on Intel:
+    # https://en.wikipedia.org/wiki/CPUID
+    return unless Hardware::CPU.intel?
+
+    # macOS `objdump` is a bit slow, so we prioritise llvm's `llvm-objdump` (~5.7x faster)
+    # or binutils' `objdump` (~1.8x faster) if they are installed.
```

Comment:
Do these times include the installation time of `llvm` or `binutils`? Even if so: I'm not sure it's worth installing `binutils` just for this rare edge-case (but could be convinced otherwise) given the potential to mess with builds.

Reply:
They do not include installation time -- this is the time difference reported from running `hyperfine` on `[llvm-]objdump -d libopenblas.dylib`.

I'm fine with returning an audit failure if there is no `objdump` on the system -- our macOS runners have it, and I believe our Linux runners do as well.

Reply:
I think it's worth benchmarking the installation time as part of the "faster" times. Otherwise it's not a great comparison.



> I'm fine with returning an audit failure if there is no `objdump` on the system -- our macOS runners have it, and I believe our Linux runners do as well.

🆒.

Reply:
> I think it's worth benchmarking the installation time as part of the "faster" times.

I think benchmarking this would be important if we were installing something in this audit, but we no longer are. I agree that it would be a useful benchmark to have, though -- especially if we end up considering installing something for this audit down the road.

Reply:
@carlocab Fine with me 👍🏻